eng/tools/typespec-validation/src/rules/compile.ts

```diff
@@ -14,21 +17,127 @@ export class CompileRule implements Rule {
 
     if (await host.checkFileExists(path.join(folder, "main.tsp"))) {
       let [err, stdout, stderr] = await host.runCmd(
-        `npm exec --no -- tsp compile --warn-as-error ${folder}`,
+        `npm exec --no -- tsp compile --list-files --warn-as-error ${folder}`,
       );
+
+      stdOutput += stdout;
+
+      // Rule output is easier to read if "tsp compile" stderr is redirected to stdOutput
+      stdOutput += stderr;
+
       if (
         stdout.toLowerCase().includes("no emitter was configured") ||
         stdout.toLowerCase().includes("no output was generated")
       ) {
         success = false;
         errorOutput += "No emitter was configured and/or no output was generated.";
       }
-      if (err) {
-        success = false;
-        errorOutput += err.message;
+
+      if (success) {
+        if (!err) {
+          // Check for *extra* typespec-generated swagger files under the output folder, which
+          // indicates a mismatch between TypeSpec and swaggers.
+
+          // Example 'stdout':
+          //
+          // TypeSpec compiler v0.67.2
+          //
+          // ../resource-manager/Microsoft.Contoso/stable/2021-11-01/contoso.json
+          // ../resource-manager/Microsoft.Contoso/stable/2021-11-01/examples/Operations_List.json
+          //
+          // Compilation completed successfully.
+
+          // Remove ANSI color codes, handle windows and linux line endings
+          const lines = stripAnsi(stdout).split(/\r?\n/);
+
+          // TODO: Use helpers in /.github once they support platform-specific paths
+          const outputSwaggers = lines
+            // Remove leading and trailing whitespace
+            .map((l) => l.trim())
+            // Skip blank lines
+            .filter((l) => l !== "")
+            // Skip first and last lines (version and success messages)
+            .slice(1, -1)
+            // Normalize to platform-specific path
+            .map((l) => normalize(l))
```

Comment:
your choice here but I do find that its much easier to not have to deal with windows path and always working only with `/` makes it much better. Typespec should only ever return unix style path. 
We have our own internal path utils that does the logic (forked from typescript) but also the package `pathe` just does that too

Reply:
I have a bigger issue tracking this for all the engsys code in this repo: https://github.com/Azure/azure-rest-api-specs/issues/33193

My conclusion was to try to match the behavior of the Node `fs` APIs, which I believe return platform-specific paths.

Are you suggesting working internally with only posix paths?  Couldn't this create problems, unless you can guarantee all related code is also assuming this?

Reply:
OK, if this is what TypeSpec does, I'm willing to give it a shot.

Reply:
Actually, I decided to keep using platform-specific paths in engsys code for now.  It's safer since we mix a lot of languages and tools.